source/UOXJSClasses.h

```diff
@@ -20,9 +20,11 @@ static JSClass global_class =
 	JS_EnumerateStub,
 	JS_ResolveStub,
 	JS_ConvertStub,
-	JS_FinalizeStub
+	JS_FinalizeStub,
+	JSCLASS_NO_OPTIONAL_MEMBERS
```

Comment:
Agreed if that is the case. This macro sets the non-existent members to `0, 0, 0, 0, 0, 0, 0, 0`. Perhaps @DragonSlayer62, @Xoduz, or Azzerhoden (from Discord) knows if this blocks arbitrary members from being passed by dynamically-loaded JS scripts. To confirm, is there a C++ function/method elsewhere that accepts a variable number of args to populate these missing fields?

Reply:
Beyond `JSFinalizeOp`, these are mostly extraneous and obsolete `JSClass` fields that do not seem to be used in our case. I've not found an explicit reference to a field following `JS_FinalizeStub` being initialized in the UOX3 source (other than `JSCLASS_NO_OPTIONAL_MEMBERS` which is new in this branch). Do you concur given the information below, @strattond?

Reference: http://www.devdoc.net/web/developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/JSAPI_reference/JSClass.html
```
struct JSClass {
    const char          *name;
    uint32_t            flags;
    /* Optional since SpiderMonkey 37 */
    JSPropertyOp        addProperty;
    JSDeletePropertyOp  delProperty;
    JSPropertyOp        getProperty;
    JSStrictPropertyOp  setProperty;
    JSEnumerateOp       enumerate;
    JSResolveOp         resolve;
    JSConvertOp         convert;      /* Obsolete since SpiderMonkey 44 */
    /* Optional since SpiderMonkey 25 */
    JSFinalizeOp        finalize;
    /* Optional */
    JSClassInternal     reserved0;    /* Obsolete since SpiderMonkey 13 */
    JSCheckAccessOp     checkAccess;  /* Obsolete since SpiderMonkey 29 */
    JSNative            call;
    JSHasInstanceOp     hasInstance;
    JSNative            construct;
    JSXDRObjectOp       xdrObject;    /* Obsolete since SpiderMonkey 13 */
    JSTraceOp           trace;        /* Added in SpiderMonkey 17 */
    JSClassInternal     reserved1;    /* Obsolete since SpiderMonkey 13 */
    void                *reserved[N]; /* sizeof 'reserved' depends on version */
};
```

Reply:
Seems rasonable @ldilley